Makefile

```diff
@@ -328,10 +328,18 @@ protobuf proto/felixbackend.pb.go: proto/felixbackend.proto
 	              --gogofaster_out=plugins=grpc:. \
 	              felixbackend.proto
 
-bpf/xdp/generated/xdp.o: bpf/xdp/filter.c
-	mkdir -p bpf/xdp/generated
+BPF_INC_FILES := bpf/include/bpf.h
+BPF_XDP_INC_FILES :=
+
+CLANG_BUILDER_STAMP := .built-bpf-clang-builder-$(BUILDARCH)
+
+$(CLANG_BUILDER_STAMP): docker-build-images/bpf-clang-builder.Dockerfile.$(BUILDARCH)
 	# the bpf object file is not arch dependent, so we can build with the current ARCH
 	docker build -t calico-build/bpf-clang -f docker-build-images/bpf-clang-builder.Dockerfile.$(BUILDARCH) .
```

Comment:
Did you add this extra machinery because the docker command was slow?  I think you can speed it up a lot by changing `.` to `docker-build-images` so that it no longer sends the whole felix dir as build context.

Reply:
Yeah, that was the reason. Changing the directory indeed speeds that up greatly. Would you prefer to get rid of this extra machinery then?